src/parser/parser.mly

```diff
@@ -219,11 +215,12 @@ plain_simple_term:
                                                           | _ -> Tuple lst }
   | LBRACE lst=separated_list(COMMA, structure_clause) RBRACE
                                                         { Structure lst }
+  | LBRACE x=var_name WITH cs=constraint_clauses RBRACE { Signature (x,cs) }
```

Comment:
Can't we do nested sharing? Instead of `x=var_name` I would expect a computation here.